<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Lesson Title: Data cleaning and DE analysis</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Carpentries Incubator" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../datacleaning-2.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Lesson Title
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Lesson Title
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Lesson Title
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 67%" class="percentage">
    67%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 67%" aria-valuenow="67" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../datacleaning-2.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="diann.html">1. Working with DIA-NN</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="datacleaning.html">2. Exploring the dataset on PRIDE</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Data cleaning and DE analysis
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#load-processed-data-and-additional-files">Load processed data and additional files</a></li>
<li><a href="#annotating-and-filtering-our-dataset">Annotating and filtering our dataset</a></li>
<li><a href="#subset-comparison-ctrl-vs-auc">Subset Comparison: Ctrl vs aUC</a></li>
<li><a href="#differential-expression-analysis">Differential Expression Analysis</a></li>
<li><a href="#create-and-explore-contrasts">Create and Explore Contrasts</a></li>
<li><a href="#example-auc-vs-ctrl-comparison">Example: aUC vs Ctrl Comparison</a></li>
<li><a href="#visualize-top-significant-proteins">Visualize Top Significant Proteins</a></li>
<li><a href="#volcano-plot">Volcano plot</a></li>
<li><a href="#network-and-enrichment-analysis">Network and Enrichment Analysis</a></li>
<li><a href="#potential-next-steps">Potential next steps</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr></ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/datacleaning.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/datacleaning.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Exploring the
        </a>
        <a class="chapter-link float-end" href="../instructor/index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Data cleaning and DE analysis</h1>
        <p>Last updated on 2025-10-21 |

        <a href="https://example.com/template-source/lesson-example/edit/main/episodes/datacleaning-2.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What is DE analysis</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>To do DE analysis</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="load-processed-data-and-additional-files">Load processed data and additional files<a class="anchor" aria-label="anchor" href="#load-processed-data-and-additional-files"></a></h2>
<hr class="half-width"><p>This data was processed using DIA-NN in the method described in the
paper.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Getting the subset of the data</span></span>
<span></span>
<span><span class="co"># curl::curl_download('https://ftp.pride.ebi.ac.uk/pride/data/archive/2024/02/PXD047585/report.tsv','data/report.tsv')</span></span>
<span><span class="co"># report_tsv &lt;- read.csv('data/report.tsv', sep='\t')</span></span>
<span><span class="co"># write.csv(stool_tsv_test,'~data/report.csv')</span></span>
<span><span class="co"># parquetize::csv_to_parquet('data/report.csv',path_to_parquet = 'data/report.parquet')</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># report_full &lt;- arrow::read_parquet('data/report.parquet')</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># proteins_keep &lt;- c("RED2_HUMAN', 'MINT_HUMAN', 'Cationic', 'OVCH2_HUMAN', 'ITA4_HUMAN', 'Trypsin', 'G3PT_HUMAN', 'ZN451_HUMAN', 'DCD_HUMAN', 'CDHR2_HUMAN', 'DSG1_HUMAN', 'BRCA2_HUMAN', 'SIAE_HUMAN', 'CTRC_HUMAN', 'CEL3A_HUMAN', 'A1AT_HUMAN', 'CYTA_HUMAN', 'CADH1_HUMAN', 'TRY2_HUMAN', 'CEAM7_HUMAN', 'ITLN1_HUMAN', 'CEL3B_HUMAN', 'DPEP1_HUMAN', 'CUZD1_HUMAN', 'ZG16_HUMAN', 'ENPP7_HUMAN', 'HV601_HUMAN', 'SMR3B_HUMAN', 'PA21B_HUMAN', 'FCGBP_HUMAN', 'LEG4_HUMAN', 'RN170_HUMAN', 'IGLL5_HUMAN', 'PPBI_HUMAN', 'CLD34_HUMAN', 'CEAM5_HUMAN', 'CEAM6_HUMAN', 'MGA_HUMAN', 'TFF3_HUMAN', 'ENTK_HUMAN', 'IGKC_HUMAN', 'HV307_HUMAN', 'IGJ_HUMAN', 'CBPB1_HUMAN', 'KV401_HUMAN', 'CTRB2_HUMAN', 'CBPA1_HUMAN', 'AMYP_HUMAN', 'SODC_HUMAN', 'CEL2A_HUMAN")</span></span>
<span><span class="co"># report_small &lt;- report_full %&gt;%</span></span>
<span><span class="co">#   filter(Protein.Group %in% proteins_keep | runif(n()) &lt; 0.10)</span></span>
<span><span class="co"># arrow::write_parquet(report_small, "data/report_subset.parquet")</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load peptide-level data processed by DIA-NN</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">readDIANN</span><span class="op">(</span>file<span class="op">=</span><span class="st">'data/report_subset.parquet'</span>,format<span class="op">=</span><span class="st">"parquet"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">names</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "E"     "genes"</code></pre>
</div>
</section><section><h2 class="section-heading" id="annotating-and-filtering-our-dataset">Annotating and filtering our dataset<a class="anchor" aria-label="anchor" href="#annotating-and-filtering-our-dataset"></a></h2>
<hr class="half-width"><p>In order to do differential expression analysis, we need information
about what samples were used.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The SampleAnnotation file was downloaded directly from PRIDE using this link&gt;</span></span>
<span><span class="co"># curl::curl_download(url='https://ftp.pride.ebi.ac.uk/pride/data/archive/2024/02/PXD047585/SampleAnnotation.xlsx',destfile = 'data/SampleAnnotation.xlsx')</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If you are on a Mac, this data needed to be cleaned up a bit with gsub if you download directly from PRIDE</span></span>
<span><span class="va">samples_stool</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">readxl</span><span class="fu">::</span><span class="fu">read_excel</span><span class="op">(</span><span class="st">'data/SampleAnnotation.xlsx'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">"\\\\"</span>, <span class="st">"/"</span>, <span class="va">samples_stool</span><span class="op">$</span><span class="va">file_name</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span> <span class="op">&lt;-</span> <span class="fu">basename</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">".mzML"</span>, <span class="st">""</span>,<span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span><span class="op">)</span></span></code></pre>
</div>
<p>The contaminant information is on GitHub and can be downloaded from
their GitHub page.</p>
<p>More info available here: Frankenfield AM, Ni J, Ahmed M, Hao L.
Protein contaminants matter: building universal protein contaminant
libraries for DDA and DIA proteomics. Journal of proteome research. 2022
Jul 6;21(9):2104-13. <a href="https://pubs.acs.org/doi/full/10.1021/acs.jproteome.2c00145" class="external-link uri">https://pubs.acs.org/doi/full/10.1021/acs.jproteome.2c00145</a></p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Code for downloading the contaminant libraries</span></span>
<span></span>
<span><span class="co"># curl::curl_download('https://github.com/HaoGroup-ProtContLib/Protein-Contaminant-Libraries-for-DDA-and-DIA-Proteomics/raw/refs/heads/main/Universal%20protein%20contaminant%20FASTA/Contaminant%20protein%20information%20in%20the%20FASTA%20library%20and%20the%20potential%20source%20of%20contaminations.xlsx',destfile = 'data/Contaminants.xlsx')</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contaminants</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">readxl</span><span class="fu">::</span><span class="fu">read_excel</span><span class="op">(</span><span class="st">'data/Contaminants.xlsx'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>New names:
• `` -&gt; `...2`
• `` -&gt; `...3`
• `` -&gt; `...4`
• `` -&gt; `...5`
• `` -&gt; `...6`
• `` -&gt; `...7`
• `` -&gt; `...8`</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contaminants_colnames</span> <span class="op">&lt;-</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">contaminants</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">contaminants</span> <span class="op">&lt;-</span> <span class="va">contaminants</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fu">nrow</span><span class="op">(</span><span class="va">contaminants</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">contaminants</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">contaminants_colnames</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check how many proteins match contaminants</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Protein.Group</span> <span class="op">%in%</span> <span class="va">contaminants</span><span class="op">$</span><span class="va">`Uniprot ID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
FALSE  TRUE
 6202   313 </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out contaminants</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span>, <span class="op">!</span><span class="op">(</span><span class="va">Protein.Group</span> <span class="op">%in%</span> <span class="va">contaminants</span><span class="op">$</span><span class="va">`Uniprot ID`</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Confirm removal</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Protein.Group</span> <span class="op">%in%</span> <span class="va">contaminants</span><span class="op">$</span><span class="va">`Uniprot ID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span></span>
<span><span class="cn">FALSE</span></span>
<span> <span class="fl">6202</span> </span></code></pre>
</div>
<div class="section level3">
<h3 id="align-sample-metadata">Align sample metadata<a class="anchor" aria-label="anchor" href="#align-sample-metadata"></a></h3>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">samples_stool</span><span class="op">[</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">%in%</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">sample_name</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">[</span>,<span class="fu">rownames</span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add experimental metadata</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'Class'</span>, <span class="st">'Batch'</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Apply peptide-level filters</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterNonProteotypicPeptides</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterCompoundProteins</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterSingletonPeptides</span><span class="op">(</span><span class="va">y.peptide</span>, min.n.peptides <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<p>For more information on the peptide-level filters, please consult the
limpa vignette.</p>
</div>
<div class="section level3">
<h3 id="dimensionality-reduction-and-qc">Dimensionality Reduction and QC<a class="anchor" aria-label="anchor" href="#dimensionality-reduction-and-qc"></a></h3>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dpcfit</span> <span class="op">&lt;-</span> <span class="fu">dpc</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>157 peptides are completely missing in all samples.</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotDPC</span><span class="op">(</span><span class="va">dpcfit</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>By protein:</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.protein</span> <span class="op">&lt;-</span> <span class="fu">dpcQuant</span><span class="op">(</span><span class="va">y.peptide</span>, <span class="st">"Protein.Names"</span>, dpc<span class="op">=</span><span class="va">dpcfit</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Estimating hyperparameters ...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Quantifying proteins ...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Proteins: 406 Peptides: 4035</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>That plot is extremely messy and doesn’t tell us much! Let’s clean it
up.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Class visualisation</span></span>
<span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Class</span><span class="op">)</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Ctrl"</span>,<span class="st">"aCD"</span>,<span class="st">"aUC"</span>,<span class="st">"CDr"</span>,<span class="st">"UCr"</span><span class="op">)</span></span>
<span><span class="va">Class.color</span> <span class="op">&lt;-</span> <span class="va">Class</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"pink"</span>,<span class="st">"black"</span>,<span class="st">'grey'</span>,<span class="st">'darkorange'</span>,<span class="st">'darkgreen'</span><span class="op">)</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Hmmm, not a lot of differences. Let’s see if there are any batch
effects.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Batch visualisation</span></span>
<span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span>
<span><span class="va">Batch.color</span> <span class="op">&lt;-</span> <span class="va">Batch</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"green"</span>,<span class="st">"purple"</span>,<span class="st">'orange'</span><span class="op">)</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It looks like there are some differences between the batches. Let’s
remove the batch effects and see how it looks.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Correct for batch effects</span></span>
<span><span class="va">y.protein.rbe</span> <span class="op">&lt;-</span> <span class="fu">removeBatchEffect</span><span class="op">(</span><span class="va">y.protein</span>,batch <span class="op">=</span> <span class="va">y.protein</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>design matrix of interest not specified. Assuming a one-group experiment.</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMDS</span><span class="op">(</span><span class="va">y.protein.rbe</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-15-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></div>
</section><section><h2 class="section-heading" id="subset-comparison-ctrl-vs-auc">Subset Comparison: Ctrl vs aUC<a class="anchor" aria-label="anchor" href="#subset-comparison-ctrl-vs-auc"></a></h2>
<hr class="half-width"><div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cdr_bool</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Class</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Ctrl'</span>,<span class="st">'aUC'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span><span class="op">[</span>, <span class="va">cdr_bool</span><span class="op">]</span>, </span>
<span>                pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">[</span><span class="va">cdr_bool</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-16-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMDS</span><span class="op">(</span><span class="va">y.protein</span><span class="op">[</span>, <span class="va">cdr_bool</span><span class="op">]</span>, </span>
<span>         pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">[</span><span class="va">cdr_bool</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-16-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMDS</span><span class="op">(</span><span class="va">y.protein.rbe</span><span class="op">[</span>, <span class="va">cdr_bool</span><span class="op">]</span>, </span>
<span>         pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">[</span><span class="va">cdr_bool</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-16-3.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We should add batch as a covariate to our analysis.</p>
</section><section><h2 class="section-heading" id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a></h2>
<hr class="half-width"><p>We now fit a linear model that includes both Class and Batch
effects.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="va">Class</span> <span class="op">+</span> <span class="va">Batch</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">dpcDE</span><span class="op">(</span><span class="va">y.protein</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-17-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">eBayes</span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">"ClassCtrl"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 Protein.Group      Protein.Names
DPP2_HUMAN                              Q9UHL4         DPP2_HUMAN
Beta-galactosidase contam_sp|P00722|BGAL_ECOLI Beta-galactosidase
PSME2_HUMAN                             Q9UL46        PSME2_HUMAN
IGLC7_HUMAN                             A0M8Q6        IGLC7_HUMAN
GPA33_HUMAN                             Q99795        GPA33_HUMAN
MYO1D_HUMAN                             O94832        MYO1D_HUMAN
PRDX1_HUMAN                             Q06830        PRDX1_HUMAN
KCRB_HUMAN                              P12277         KCRB_HUMAN
HSPB1_HUMAN                             P04792        HSPB1_HUMAN
HMGB2_HUMAN                             P26583        HMGB2_HUMAN
                                Genes NPeptides     PropObs     logFC   AveExpr
DPP2_HUMAN                       DPP7         2 0.003225806 10.220764 10.221984
Beta-galactosidase Beta-galactosidase         2 0.006451613  9.924804  9.923592
PSME2_HUMAN                     PSME2         2 0.006451613 10.226636 10.228126
IGLC7_HUMAN                     IGLC7         2 0.006451613 10.048682 10.047295
GPA33_HUMAN                     GPA33         3 0.004301075  9.660197  9.662022
MYO1D_HUMAN                     MYO1D         2 0.009677419  9.731588  9.728144
PRDX1_HUMAN                     PRDX1         2 0.009677419 10.432375 10.436082
KCRB_HUMAN                        CKB         2 0.012903226 10.242352 10.244198
HSPB1_HUMAN                     HSPB1         2 0.006451613 10.491556 10.495906
HMGB2_HUMAN                     HMGB2         2 0.003225806 10.312915 10.316145
                          t P.Value adj.P.Val        B
DPP2_HUMAN         4877.269       0         0 850.6721
Beta-galactosidase 6395.855       0         0 841.4553
PSME2_HUMAN        6597.852       0         0 838.1056
IGLC7_HUMAN        3960.205       0         0 830.1724
GPA33_HUMAN        3963.989       0         0 827.7626
MYO1D_HUMAN        4831.810       0         0 826.2010
PRDX1_HUMAN        5230.944       0         0 816.8051
KCRB_HUMAN         4244.864       0         0 806.9717
HSPB1_HUMAN        3998.250       0         0 805.8680
HMGB2_HUMAN        2999.040       0         0 800.8352</code></pre>
</div>
</section><section><h2 class="section-heading" id="create-and-explore-contrasts">Create and Explore Contrasts<a class="anchor" aria-label="anchor" href="#create-and-explore-contrasts"></a></h2>
<hr class="half-width"><p>We can define custom contrasts (e.g., comparing aCD vs Ctrl) and
explore differential proteins.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define contrasts</span></span>
<span><span class="va">contrasts_fit</span> <span class="op">&lt;-</span> <span class="fu">makeContrasts</span><span class="op">(</span>Ctrl_aCD <span class="op">=</span> <span class="va">ClassaCD</span> <span class="op">-</span> <span class="va">ClassCtrl</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Apply contrasts</span></span>
<span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu">contrasts.fit</span><span class="op">(</span><span class="va">fit</span>, contrasts <span class="op">=</span> <span class="va">contrasts_fit</span><span class="op">)</span></span>
<span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu">eBayes</span><span class="op">(</span><span class="va">fit2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summaries</span></span>
<span><span class="fu">topTable</span><span class="op">(</span><span class="va">fit2</span>, coef <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>            Protein.Group Protein.Names    Genes NPeptides    PropObs
ALBU_HUMAN         P02768    ALBU_HUMAN      ALB        81 0.02692155
HBB_HUMAN          P68871     HBB_HUMAN      HBB        11 0.02697947
S10A9_HUMAN        P06702   S10A9_HUMAN   S100A9        30 0.05612903
DPP4_HUMAN         P27487    DPP4_HUMAN     DPP4        54 0.05627240
NEP_HUMAN          P08473     NEP_HUMAN      MME        49 0.05345622
TRFE_HUMAN         P02787    TRFE_HUMAN       TF        32 0.01108871
HEMO_HUMAN         P02790    HEMO_HUMAN      HPX         9 0.01505376
MUC2_HUMAN         Q02817    MUC2_HUMAN     MUC2        79 0.04752960
ANT3_HUMAN         P01008    ANT3_HUMAN SERPINC1        34 0.04724858
PRTN3_HUMAN        P24158   PRTN3_HUMAN    PRTN3        10 0.04709677
                  logFC  AveExpr         t      P.Value   adj.P.Val         B
ALBU_HUMAN   2.23879844 10.40131  4.478478 1.480449e-05 0.006010624  1.677531
HBB_HUMAN    0.36899249 11.35499  2.357978 1.966516e-02 0.998947663 -3.357132
S10A9_HUMAN -0.67889289 12.45692 -2.174909 3.120371e-02 0.998947663 -3.530783
DPP4_HUMAN   0.76076855 11.94575  2.030031 4.412074e-02 0.998947663 -3.771985
NEP_HUMAN    0.51280892 11.55182  1.850375 6.622674e-02 0.998947663 -4.036171
TRFE_HUMAN   0.21123633 10.12176  1.963051 5.148991e-02 0.998947663 -4.251168
HEMO_HUMAN   0.08091332 11.54930  1.974498 5.016131e-02 0.998947663 -4.262076
MUC2_HUMAN   0.66751339 11.82464  1.630523 1.050890e-01 0.998947663 -4.262562
ANT3_HUMAN  -0.31282593 10.81556 -1.740333 8.385128e-02 0.998947663 -4.525959
PRTN3_HUMAN -0.29640473 11.81750 -1.691011 9.291138e-02 0.998947663 -4.634509</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare multiple coefficients</span></span>
<span><span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 Protein.Group      Protein.Names
DPP2_HUMAN                              Q9UHL4         DPP2_HUMAN
Beta-galactosidase contam_sp|P00722|BGAL_ECOLI Beta-galactosidase
PSME2_HUMAN                             Q9UL46        PSME2_HUMAN
IGLC7_HUMAN                             A0M8Q6        IGLC7_HUMAN
GPA33_HUMAN                             Q99795        GPA33_HUMAN
MYO1D_HUMAN                             O94832        MYO1D_HUMAN
PRDX1_HUMAN                             Q06830        PRDX1_HUMAN
KCRB_HUMAN                              P12277         KCRB_HUMAN
HSPB1_HUMAN                             P04792        HSPB1_HUMAN
HMGB2_HUMAN                             P26583        HMGB2_HUMAN
                                Genes NPeptides     PropObs     logFC   AveExpr
DPP2_HUMAN                       DPP7         2 0.003225806 10.220764 10.221984
Beta-galactosidase Beta-galactosidase         2 0.006451613  9.924804  9.923592
PSME2_HUMAN                     PSME2         2 0.006451613 10.226636 10.228126
IGLC7_HUMAN                     IGLC7         2 0.006451613 10.048682 10.047295
GPA33_HUMAN                     GPA33         3 0.004301075  9.660197  9.662022
MYO1D_HUMAN                     MYO1D         2 0.009677419  9.731588  9.728144
PRDX1_HUMAN                     PRDX1         2 0.009677419 10.432375 10.436082
KCRB_HUMAN                        CKB         2 0.012903226 10.242352 10.244198
HSPB1_HUMAN                     HSPB1         2 0.006451613 10.491556 10.495906
HMGB2_HUMAN                     HMGB2         2 0.003225806 10.312915 10.316145
                          t P.Value adj.P.Val        B
DPP2_HUMAN         4877.269       0         0 850.6721
Beta-galactosidase 6395.855       0         0 841.4553
PSME2_HUMAN        6597.852       0         0 838.1056
IGLC7_HUMAN        3960.205       0         0 830.1724
GPA33_HUMAN        3963.989       0         0 827.7626
MYO1D_HUMAN        4831.810       0         0 826.2010
PRDX1_HUMAN        5230.944       0         0 816.8051
KCRB_HUMAN         4244.864       0         0 806.9717
HSPB1_HUMAN        3998.250       0         0 805.8680
HMGB2_HUMAN        2999.040       0         0 800.8352</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 Protein.Group      Protein.Names
DPP2_HUMAN                              Q9UHL4         DPP2_HUMAN
Beta-galactosidase contam_sp|P00722|BGAL_ECOLI Beta-galactosidase
PSME2_HUMAN                             Q9UL46        PSME2_HUMAN
IGLC7_HUMAN                             A0M8Q6        IGLC7_HUMAN
GPA33_HUMAN                             Q99795        GPA33_HUMAN
MYO1D_HUMAN                             O94832        MYO1D_HUMAN
PRDX1_HUMAN                             Q06830        PRDX1_HUMAN
HSPB1_HUMAN                             P04792        HSPB1_HUMAN
SCMC1_HUMAN                             Q6NUK1        SCMC1_HUMAN
ACTN3_HUMAN                             Q08043        ACTN3_HUMAN
                                Genes NPeptides     PropObs     logFC   AveExpr
DPP2_HUMAN                       DPP7         2 0.003225806 10.220752 10.221984
Beta-galactosidase Beta-galactosidase         2 0.006451613  9.925068  9.923592
PSME2_HUMAN                     PSME2         2 0.006451613 10.226641 10.228126
IGLC7_HUMAN                     IGLC7         2 0.006451613 10.047665 10.047295
GPA33_HUMAN                     GPA33         3 0.004301075  9.661512  9.662022
MYO1D_HUMAN                     MYO1D         2 0.009677419  9.731603  9.728144
PRDX1_HUMAN                     PRDX1         2 0.009677419 10.432355 10.436082
HSPB1_HUMAN                     HSPB1         2 0.006451613 10.491542 10.495906
SCMC1_HUMAN                  SLC25A24         2 0.009677419 10.453710 10.457892
ACTN3_HUMAN                     ACTN3         2 0.006451613 10.381462 10.385495
                          t P.Value adj.P.Val        B
DPP2_HUMAN         6014.446       0         0 882.1088
Beta-galactosidase 7930.433       0         0 873.7109
PSME2_HUMAN        8137.383       0         0 869.5640
IGLC7_HUMAN        4972.912       0         0 864.3338
GPA33_HUMAN        4951.280       0         0 861.1143
MYO1D_HUMAN        5887.336       0         0 855.8380
PRDX1_HUMAN        6446.345       0         0 848.1433
HSPB1_HUMAN        4930.013       0         0 837.2906
SCMC1_HUMAN        5176.668       0         0 835.4328
ACTN3_HUMAN        4544.208       0         0 833.0233</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 Protein.Group      Protein.Names
DPP2_HUMAN                              Q9UHL4         DPP2_HUMAN
Beta-galactosidase contam_sp|P00722|BGAL_ECOLI Beta-galactosidase
PSME2_HUMAN                             Q9UL46        PSME2_HUMAN
IGLC7_HUMAN                             A0M8Q6        IGLC7_HUMAN
GPA33_HUMAN                             Q99795        GPA33_HUMAN
MYO1D_HUMAN                             O94832        MYO1D_HUMAN
PRDX1_HUMAN                             Q06830        PRDX1_HUMAN
KCRB_HUMAN                              P12277         KCRB_HUMAN
HSPB1_HUMAN                             P04792        HSPB1_HUMAN
ACTN3_HUMAN                             Q08043        ACTN3_HUMAN
                                Genes NPeptides     PropObs     logFC   AveExpr
DPP2_HUMAN                       DPP7         2 0.003225806 10.222675 10.221984
Beta-galactosidase Beta-galactosidase         2 0.006451613  9.925129  9.923592
PSME2_HUMAN                     PSME2         2 0.006451613 10.226647 10.228126
IGLC7_HUMAN                     IGLC7         2 0.006451613 10.046443 10.047295
GPA33_HUMAN                     GPA33         3 0.004301075  9.663383  9.662022
MYO1D_HUMAN                     MYO1D         2 0.009677419  9.730722  9.728144
PRDX1_HUMAN                     PRDX1         2 0.009677419 10.433241 10.436082
KCRB_HUMAN                        CKB         2 0.012903226 10.242268 10.244198
HSPB1_HUMAN                     HSPB1         2 0.006451613 10.492542 10.495906
ACTN3_HUMAN                     ACTN3         2 0.006451613 10.381636 10.385495
                          t P.Value adj.P.Val        B
DPP2_HUMAN         4919.450       0         0 851.9517
Beta-galactosidase 6649.977       0         0 847.2963
PSME2_HUMAN        6749.099       0         0 841.5052
IGLC7_HUMAN        4064.202       0         0 834.0719
GPA33_HUMAN        4044.728       0         0 830.7692
MYO1D_HUMAN        4826.051       0         0 826.0299
PRDX1_HUMAN        5213.601       0         0 816.2985
KCRB_HUMAN         4266.358       0         0 807.7300
HSPB1_HUMAN        4006.021       0         0 806.1518
ACTN3_HUMAN        3763.088       0         0 804.7304</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 Protein.Group      Protein.Names    Genes
DIGEST_Label13C15N                    P9999999 DIGEST_Label13C15N DL13C15N
Trypsin              contam_sp|P00761|TRYP_PIG            Trypsin  Trypsin
CTRC_HUMAN                              Q99895         CTRC_HUMAN     CTRC
MGA_HUMAN                               O43451          MGA_HUMAN     MGAM
CEL3A_HUMAN                             P09093        CEL3A_HUMAN   CELA3A
CDHR2_HUMAN                             Q9BYE9        CDHR2_HUMAN    CDHR2
CBPA1_HUMAN                             P15085        CBPA1_HUMAN     CPA1
DPEP1_HUMAN                             P16444        DPEP1_HUMAN    DPEP1
Serum              contam_sp|P02769|ALBU_BOVIN              Serum    Serum
MUC2_HUMAN                              Q02817         MUC2_HUMAN     MUC2
                   NPeptides    PropObs      logFC  AveExpr         t
DIGEST_Label13C15N        14 0.02672811  0.4528090 12.80679  5.756188
Trypsin                   26 0.05707196 -0.5672180 13.66182 -4.054078
CTRC_HUMAN                37 0.07061901 -0.7645132 14.23116 -3.872798
MGA_HUMAN                 92 0.06367461 -0.7376365 12.84200 -3.460817
CEL3A_HUMAN               34 0.06907021 -0.5811737 14.01516 -3.399523
CDHR2_HUMAN               40 0.08016129 -0.5862240 13.28108 -3.170783
CBPA1_HUMAN               49 0.05543120 -0.6468480 12.69218 -2.109360
DPEP1_HUMAN               27 0.06905615 -0.2534077 12.33003 -1.925096
Serum                     53 0.03992696  0.3432754 10.74604  1.803974
MUC2_HUMAN                79 0.04752960  0.4196754 11.82464  1.599716
                        P.Value    adj.P.Val          B
DIGEST_Label13C15N 4.693993e-08 1.905761e-05  8.2068985
Trypsin            8.058659e-05 1.635908e-02  1.2767389
CTRC_HUMAN         1.602282e-04 2.168421e-02  0.6739633
MGA_HUMAN          7.016031e-04 7.023562e-02 -0.5608611
CEL3A_HUMAN        8.649707e-04 7.023562e-02 -0.8696406
CDHR2_HUMAN        1.843621e-03 1.247517e-01 -1.4593639
CBPA1_HUMAN        3.657416e-02 9.999636e-01 -4.1158372
DPEP1_HUMAN        5.610980e-02 9.999636e-01 -4.6781628
Serum              7.324194e-02 9.999636e-01 -4.7376792
MUC2_HUMAN         1.117658e-01 9.999636e-01 -4.7700018</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example diagnostic plots</span></span>
<span><span class="fu">plotMD</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-20-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMD</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-20-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="example-auc-vs-ctrl-comparison">Example: aUC vs Ctrl Comparison<a class="anchor" aria-label="anchor" href="#example-auc-vs-ctrl-comparison"></a></h2>
<hr class="half-width"><div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">3</span>, number <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize a specific protein</span></span>
<span><span class="fu">plotProtein</span><span class="op">(</span><span class="va">y.protein</span>, <span class="st">"S10A9_HUMAN"</span>, col <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'topleft'</span>, legend <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-21-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="visualize-top-significant-proteins">Visualize Top Significant Proteins<a class="anchor" aria-label="anchor" href="#visualize-top-significant-proteins"></a></h2>
<hr class="half-width"><p>We identify the top 50 most significant and variable proteins, then
visualize via a clustered heatmap.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_proteins</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">top_n</span><span class="op">(</span><span class="fl">50</span>, wt <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Protein.Names</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expr_matrix</span> <span class="op">&lt;-</span> <span class="va">y.protein</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="va">sig_proteins</span>, <span class="op">]</span></span>
<span><span class="va">scaled_expr</span> <span class="op">&lt;-</span> <span class="fu">t</span><span class="op">(</span><span class="fu">scale</span><span class="op">(</span><span class="fu">t</span><span class="op">(</span><span class="va">expr_matrix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">col_annotation</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>Class <span class="op">=</span> <span class="va">Class</span>,</span>
<span>                             row.names <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">y.protein</span><span class="op">$</span><span class="va">E</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span><span class="va">scaled_expr</span>,</span>
<span>         cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         show_rownames <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         show_colnames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>         annotation_col <span class="op">=</span> <span class="va">col_annotation</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-22-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a></h2>
<hr class="half-width"><div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EnhancedVolcano</span><span class="op">(</span><span class="va">results</span>,</span>
<span>                lab <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">Protein.Names</span>,</span>
<span>                x <span class="op">=</span> <span class="st">'logFC'</span>,</span>
<span>                y <span class="op">=</span> <span class="st">'adj.P.Val'</span>,</span>
<span>                pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                FCcutoff <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>                pointSize <span class="op">=</span> <span class="fl">3.0</span>,</span>
<span>                labSize <span class="op">=</span> <span class="fl">3.0</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: One or more p-values is 0. Converting to 10^-1 * current lowest
non-zero p-value...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
ℹ The deprecated feature was likely used in the EnhancedVolcano package.
  Please report the issue to the authors.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.
ℹ The deprecated feature was likely used in the EnhancedVolcano package.
  Please report the issue to the authors.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_vline()`).</code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-23-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="network-and-enrichment-analysis">Network and Enrichment Analysis<a class="anchor" aria-label="anchor" href="#network-and-enrichment-analysis"></a></h2>
<hr class="half-width"><div class="section level3">
<h3 id="string-proteinprotein-interaction-network">STRING Protein–Protein Interaction Network<a class="anchor" aria-label="anchor" href="#string-proteinprotein-interaction-network"></a></h3>
<p>Note: this part does not seem to be working</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">string_db</span> <span class="op">&lt;-</span> <span class="va">STRINGdb</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>version <span class="op">=</span> <span class="st">"11.5"</span>, species <span class="op">=</span> <span class="fl">9606</span>, score_threshold <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span><span class="va">mapped</span> <span class="op">&lt;-</span> <span class="va">string_db</span><span class="op">$</span><span class="fu">map</span><span class="op">(</span><span class="va">results</span>, <span class="st">"Protein.Names"</span>, removeUnmappedRows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">de_proteins</span> <span class="op">&lt;-</span> <span class="va">mapped</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot network of significant proteins</span></span>
<span><span class="va">string_db</span><span class="op">$</span><span class="fu">plot_network</span><span class="op">(</span><span class="va">de_proteins</span><span class="op">$</span><span class="va">STRING_id</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="functional-enrichment-go-and-kegg">Functional Enrichment (GO and KEGG)<a class="anchor" aria-label="anchor" href="#functional-enrichment-go-and-kegg"></a></h3>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_proteins</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert UniProt IDs to Entrez IDs for enrichment</span></span>
<span><span class="va">converted</span> <span class="op">&lt;-</span> <span class="fu">bitr</span><span class="op">(</span><span class="va">de_proteins</span><span class="op">$</span><span class="va">Protein.Group</span>,</span>
<span>                  fromType <span class="op">=</span> <span class="st">"UNIPROT"</span>,</span>
<span>                  toType <span class="op">=</span> <span class="st">"ENTREZID"</span>,</span>
<span>                  OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in bitr(de_proteins$Protein.Group, fromType = "UNIPROT", toType =
"ENTREZID", : 16.5% of input gene IDs are fail to map...</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GO Biological Process enrichment</span></span>
<span><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">converted</span><span class="op">$</span><span class="va">ENTREZID</span>,</span>
<span>                OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,</span>
<span>                ont <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                readable <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dotplot</span><span class="op">(</span><span class="va">ego</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-25-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># KEGG pathway enrichment</span></span>
<span><span class="va">ekegg</span> <span class="op">&lt;-</span> <span class="fu">enrichKEGG</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">converted</span><span class="op">$</span><span class="va">ENTREZID</span>,</span>
<span>                    organism <span class="op">=</span> <span class="st">'hsa'</span>,</span>
<span>                    pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Reading KEGG annotation online: "https://rest.kegg.jp/link/hsa/pathway"...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/hsa"...</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dotplot</span><span class="op">(</span><span class="va">ekegg</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/datacleaning-2-rendered-unnamed-chunk-25-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></div>
</section><section><h2 class="section-heading" id="potential-next-steps">Potential next steps<a class="anchor" aria-label="anchor" href="#potential-next-steps"></a></h2>
<hr class="half-width"><ul><li><p>Compare coefficients between classes as an exercise.</p></li>
<li><p>Visualize selected proteins of interest across experimental
groups.</p></li>
<li><p>Extend enrichment analyses using Reactome or GSEA
approaches.</p></li>
<li><p>Integrate sample metadata for clinical covariates or longitudinal
modeling.</p></li>
</ul><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>Data!</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/datacleaning.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/datacleaning.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Exploring the
        </a>
        <a class="chapter-link float-end" href="../instructor/index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://example.com/template-source/lesson-example/edit/main/episodes/datacleaning-2.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://example.com/template-source/lesson-example/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://example.com/template-source/lesson-example/" class="external-link">Source</a></p>
				<p><a href="https://example.com/template-source/lesson-example/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:team@carpentries.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.17.1" class="external-link">sandpaper (0.17.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.7" class="external-link">varnish (1.0.7)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://template-source.github.io/lesson-example/instructor/datacleaning-2.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "Data cleaning and DE analysis",
  "creativeWorkStatus": "active",
  "url": "https://template-source.github.io/lesson-example/instructor/datacleaning-2.html",
  "identifier": "https://template-source.github.io/lesson-example/instructor/datacleaning-2.html",
  "dateCreated": "2025-09-30",
  "dateModified": "2025-10-21",
  "datePublished": "2025-10-21"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

